internal/common/db.go

```diff
@@ -411,6 +411,70 @@ func prepareClusterDB(db *gorm.DB, eagerLoading EagerLoadingState, includeDelete
 	return db
 }
 
+func prepareClusterDBWithJoins(db *gorm.DB, includeDeleted DeleteRecordsState) *gorm.DB {
+	if includeDeleted {
+		db = db.Unscoped()
+	}
+
+	conditions := []interface{}{func(db *gorm.DB) *gorm.DB {
+		if includeDeleted {
+			return db.Unscoped()
+		}
+		return db
+	}}
+
+	networkTables := map[string]bool{
+		ClusterNetworksTable: true,
+		ServiceNetworksTable: true,
+		MachineNetworksTable: true,
+		HostsTable:           true,
+	}
+
+	for _, tableName := range ClusterSubTables {
+		if !networkTables[tableName] {
+			db = LoadTableFromDB(db, tableName, conditions...)
+		}
+	}
+
+	db = db.Preload(ClusterNetworksTable, func(db *gorm.DB) *gorm.DB {
+		baseDB := db
+		if includeDeleted {
+			baseDB = db.Unscoped()
+		}
```

Comment:
why we need that all the time? once [here](https://github.com/openshift/assisted-service/pull/7994/files#diff-01bdb7dd494494bebd086fc6d8172ab9a4da151e7fb194a40616bfba462a27e4R419-R424) is not enough?